dlib/pixel.h

```diff
@@ -962,6 +1241,84 @@ namespace dlib
             assign_pixel_helpers::assign(dest,temp);
         }
 
+        template < typename P1, typename P2 >
+        typename enable_if_c<pixel_traits<P1>::hsi && pixel_traits<P2>::lab>::type
+        assign(P1& dest, const P2& src)
+        {
+            rgb_pixel temp;
+            // convert lab value to our temp rgb pixel
+            assign_pixel_helpers::assign(temp,src);
+            // now we can just go assign the new rgb value to the
+            // hsi pixel
+            assign_pixel_helpers::assign(dest,temp);
+        }
+
+    // -----------------------------
+        // dest is an lab pixel
+        template < typename P1>
+        typename enable_if_c<pixel_traits<P1>::lab>::type
+        assign(P1& dest, const unsigned char& src)
+        {
+            dest.a = 0;
+            dest.b = 0;
+            dest.l = src;
+        }
+
+
+        template < typename P1, typename P2 >
+        typename enable_if_c<pixel_traits<P1>::lab && pixel_traits<P2>::grayscale>::type
+        assign(P1& dest, const P2& src)
+        {
+            dest.a = 0;
+            dest.b = 0;
+            assign_pixel(dest.l, src);
```

Comment:
Yes this is definitely incorrect, I will fix it when I decide how to represent the L value correctly.